reference-service/src/main/java/com/transformuk/hee/tis/reference/service/service/impl/SpecificationFactory.java

```diff
@@ -86,6 +86,34 @@ public static Specification in(String attribute, Collection<Object> values) {
     };
   }
 
+  /**
+   * In condition for entity property, if property is from sub entity then it should contain '.' e.g
+   * sites.siteId, and is made specifically for strings where case is not relevant.
+   *
+   * @param attribute table name
+   * @param values values to be searched
+   * @return Specification
+   */
+  public static Specification inIgnoreCase(String attribute, Collection<String> values) {
+    return (root, query, cb) -> {
+      CriteriaBuilder.In cbi;
+      if (StringUtils.isNotEmpty(attribute) && attribute.contains(DOT)) {
+        // this support multiple entity in criteria e.g specialties.specialty.name or sites.siteId
+        String[] joinTable = StringUtils.split(attribute, DOT);
+        Join tableJoin = root.join(joinTable[0], JoinType.INNER);
+        for (int i = 1; i < joinTable.length - 1; i++) {
+          tableJoin = tableJoin.join(joinTable[i], JoinType.INNER);
+        }
+        cbi = cb.in(cb.upper(tableJoin.get(joinTable[joinTable.length - 1]))); // attribute
+      } else {
+        cbi = cb.in(cb.upper(root.get(attribute)));
+      }
+      values.forEach(v ->
+          cbi.value(v.toUpperCase()));
```

Comment:
Comparing to the original `in` method, this line is the difference. But the `in` method already deals with case insensitivity, so `inIgnoreCase` is not required.